reviewrot/pagurestack.py

```
@@ -0,0 +1,136 @@
+                date = datetime.datetime.strptime(created_date,
+                                                  '%Y-%m-%d %H:%M:%S.%f')
+            except ValueError:
+                date = datetime.datetime.strptime(created_date,
+                                                  '%Y-%m-%d %H:%M:%S')
+            """ check if review request is older/newer than specified time
+            interval"""
+            result = self.check_request_state(date,
+                                              state_, value, duration)
+            if result is False:
+                log.debug("pull request '%s' is not %s than specified"
+                          " time interval", res['title'], state_)
+                continue
+            # format the time interval pull request has been filed since
+            time = self.format_duration(created_at=date)
+            res = PagureReview(user=res['user']['name'],
+                               title=res['title'],
+                               url=url,
+                               time=time,
+                               comments=len(res['comments']))
+            log.info(res)
+            res_.append(res)
+        return res_
+
+    def _call_api(self, url, method='GET'):
+        """
+        Method used to call the API.
+        It returns the raw JSON returned by the API or raises an exception
+        if something goes wrong.
+
+        Args:
+            method (str): the URL to call, can be GET, POST, DELETE, UPDATE...
+                          Defaults to GET
+
+        Returns:
+            raw JSON returned by API
+        """
+        req = self.session.request(
+            method=method,
+            url=url,
+            headers=self.header,
+        )
+        output = None
+        try:
+            output = req.json()
+        except Exception as err:
+            log.exception('Error while decoding JSON: {0}')
+            raise Exception('Error while decoding JSON: {0}'.format(err))
```

Also, you might as well not mask the original exception ``err`` here.

If you just write:

```python
log.exception('Error while decoding JSON: {0}'.format(err))
raise
```

It will re-raise the original exception with its original `type`.